src/vmm/src/vmm_config/logger.rs

```diff
@@ -1,257 +1,298 @@
+        if let Some(level) = self.level {
+            filter
+                .modify(|f| *f = LevelFilter::from_level(tracing::Level::from(level)))
+                .map_err(UpdateLoggerError::Filter)?;
+        }
 
-        let mut line = String::new();
-        loop {
-            if line.contains("Guest-boot-time =") {
-                break;
-            }
-            if reader.read_line(&mut line).unwrap() == 0 {
-                // If it ever gets here, this assert will fail.
-                assert!(line.contains("Guest-boot-time ="));
-            }
+        // Update if the logger shows the level
+        if let Some(show_level) = self.show_level {
+            SHOW_LEVEL.store(show_level, SeqCst);
         }
+
+        // Updates if the logger shows the origin
+        if let Some(show_log_origin) = self.show_log_origin {
+            SHOW_LOG_ORIGIN.store(show_log_origin, SeqCst);
+        }
+
+        Ok(())
     }
+}
 
-    #[test]
-    fn test_new_logger_config() {
-        let logger_config =
-            LoggerConfig::new(PathBuf::from("log"), LoggerLevel::Debug, false, true);
-        assert_eq!(logger_config.log_path, PathBuf::from("log"));
-        assert_eq!(logger_config.level, LoggerLevel::Debug);
-        assert!(!logger_config.show_level);
-        assert!(logger_config.show_log_origin);
+#[derive(Debug)]
+struct LoggerFormatter;
+impl LoggerFormatter {
+    pub fn new(show_level: bool, show_log_origin: bool) -> Self {
+        SHOW_LEVEL.store(show_level, SeqCst);
+        SHOW_LOG_ORIGIN.store(show_log_origin, SeqCst);
+        Self
     }
+}
 
-    #[test]
-    fn test_parse_level() {
-        // Check `from_string()` behaviour for different scenarios.
-        assert_eq!(
-            format!(
-                "{}",
-                LoggerLevel::from_string("random_value".to_string()).unwrap_err()
-            ),
-            "random_value"
-        );
-        assert_eq!(
-            LoggerLevel::from_string("Error".to_string()).unwrap(),
-            LoggerLevel::Error
-        );
-        assert_eq!(
-            LoggerLevel::from_string("Warning".to_string()).unwrap(),
-            LoggerLevel::Warning
-        );
-        assert_eq!(
-            LoggerLevel::from_string("Info".to_string()).unwrap(),
-            LoggerLevel::Info
-        );
-        assert_eq!(
-            LoggerLevel::from_string("Debug".to_string()).unwrap(),
-            LoggerLevel::Debug
-        );
-        assert_eq!(
-            LoggerLevel::from_string("error".to_string()).unwrap(),
-            LoggerLevel::Error
-        );
-        assert_eq!(
-            LoggerLevel::from_string("WaRnIng".to_string()).unwrap(),
-            LoggerLevel::Warning
-        );
-        assert_eq!(
-            LoggerLevel::from_string("DEBUG".to_string()).unwrap(),
-            LoggerLevel::Debug
-        );
+static SHOW_LEVEL: AtomicBool = AtomicBool::new(false);
+static SHOW_LOG_ORIGIN: AtomicBool = AtomicBool::new(false);
```

Comment:
`SHOW_LEVEL` could be updated to provide more verbose data about a specific API call a user might be interested in.

Reply:
> I think this is fine, I can't think of a circumstance where this is an issue.

It might be an issue if a user is expecting particular log formats at every given point in time.

> SHOW_LEVEL could be updated to provide more verbose data about a specific API call a user might be interested in.

I'm not sure how someone can do that per API call, based on your changes. Could you give an example?

Reply:
```bash
# Default log level is `Warn` update to `Debug` to be more verbose.
curl -X PUT --unix-socket "${API_SOCKET}" \
    --data "{
        \"level\": \"Debug\",
    }" \
    "http://localhost/logger"

# Set boot source
curl -X PUT --unix-socket "${API_SOCKET}" \
    --data "{
        \"kernel_image_path\": \"${KERNEL}\",
        \"boot_args\": \"${KERNEL_BOOT_ARGS}\"
    }" \
    "http://localhost/boot-source"

# Reduce log verbosity back down to `Warn`
curl -X PUT --unix-socket "${API_SOCKET}" \
    --data "{
        \"level\": \"Warn\",
    }" \
    "http://localhost/logger"

```